core/services/gateway/handlers/functions/subscriptions.go

```diff
@@ -203,3 +225,30 @@ func (s *onchainSubscriptions) getSubscriptionsCount(ctx context.Context, blockN
 		Context:     ctx,
 	})
 }
+
+func (s *onchainSubscriptions) loadCachedSubscriptions() {
+	offset := uint(0)
+	for {
+		csBatch, err := s.orm.GetSubscriptions(offset, s.config.CacheBatchSize)
+		if err != nil {
+			break
+		}
+
+		for _, cs := range csBatch {
+			_ = s.subscriptions.UpdateSubscription(cs.SubscriptionID, &functions_router.IFunctionsSubscriptionsSubscription{
+				Balance:        cs.Balance,
+				Owner:          cs.Owner,
+				BlockedBalance: cs.BlockedBalance,
+				ProposedOwner:  cs.ProposedOwner,
+				Consumers:      cs.Consumers,
+				Flags:          cs.Flags,
+			})
+			s.lggr.Debugw("Loading cached subscriptions", "offset", offset, "batch_length", len(csBatch))
```

Comment:
nit: should be outside of the loop, right?

Reply:
true!